### PR TITLE
Fix handling of missing G frame

### PR DESCRIPTION
**Pull request by atomgomba**
I found an edge case in handling G frames when opening a log generated by Betaflight 4.5 ([37f119cf4](https://github.com/betaflight/betaflight/commit/37f119cf4)) (please see the original log [here](https://drive.google.com/file/d/1aYOb-e28adpb-HGhbcJEKizk9qiwxICk/view?usp=sharing)). In case of a corrupt frame the parser ([flightlog_parser.js:1779](https://github.com/betaflight/blackbox-log-viewer/blob/master/js/flightlog_parser.js#L1779)) calls the callback function like this:
```javascript
//Let the caller know there was a corrupt frame (don't give them a pointer to the frame data because it is totally worthless)
if (this.onFrameReady)
    this.onFrameReady(false, null, lastFrameType.marker, frameStart, lastFrameSize);
```
Note in above code `frame` argument is `null`.

But the callee in [flightlog.js:391](https://github.com/betaflight/blackbox-log-viewer/blob/master/js/flightlog.js#L391) currently seems to be missing a condition when processing G frames to avoid reading properties of `frame` (which is `null` in this case).

Without this change I get this exception in the console when opening the log:

Screenshot from 2023-10-15 11-53-45

After applying the change in this PR everything seems to be working correctly, file opens normally and eg. GPX Export produces correct output (tested in Google Earth).

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**
